# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.71; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal; autoconf; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/bin
install rdjpgcom $DESTDIR/bin
install wrjpgcom $DESTDIR/bin
install .libs/cjpeg $DESTDIR/bin
install .libs/djpeg $DESTDIR/bin
install .libs/jpegtran $DESTDIR/bin
install -d $DESTDIR/include
install -m 644 jconfig.h $DESTDIR/include
install -m 644 jerror.h $DESTDIR/include
install -m 644 jmorecfg.h $DESTDIR/include
install -m 644 jpeglib.h $DESTDIR/include
install -m 644 transupp.h $DESTDIR/include
install -d $DESTDIR/lib
install .libs/libjpeg.so.9.* $DESTDIR/lib/libjpeg.so.9
ln -sf libjpeg.so.9 $DESTDIR/lib/libjpeg.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libjpeg.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/jpeg-build.pc << EOF
Name: jpeg
Version: 9e
Description: Joint Photographic Experts Group (JPEG) library
Requires: $REQUIRES

devel=\\
/include/jconfig.h \\
/include/jerror.h \\
/include/jmorecfg.h \\
/include/jpeglib.h \\
/include/transupp.h \\
/lib/libjpeg.so \\
/lib/pkgconfig/libjpeg.pc

exec=\\
/bin/cjpeg \\
/bin/djpeg \\
/bin/jpegtran \\
/bin/rdjpgcom \\
/bin/wrjpgcom \\
/lib/libjpeg.so.9
EOF

exit
--- jpeg-9e.orig/Makefile.am
+++ jpeg-9e/Makefile.am
@@ -13,7 +13,7 @@
         jddctmgr.c jdhuff.c jdinput.c jdmainct.c jdmarker.c jdmaster.c \
         jdmerge.c jdpostct.c jdsample.c jdtrans.c jerror.c jfdctflt.c \
         jfdctfst.c jfdctint.c jidctflt.c jidctfst.c jidctint.c jquant1.c \
-        jquant2.c jutils.c jmemmgr.c @MEMORYMGR@.c
+        jquant2.c jutils.c jmemmgr.c transupp.c @MEMORYMGR@.c
 
 # System dependent sources
 SYSDEPSOURCES = jmemansi.c jmemname.c jmemnobs.c jmemdos.c jmemmac.c
@@ -88,7 +88,7 @@
 djpeg_SOURCES    = djpeg.c wrppm.c wrgif.c wrtarga.c wrrle.c wrbmp.c \
         rdcolmap.c cdjpeg.c
 djpeg_LDADD      = libjpeg.la
-jpegtran_SOURCES = jpegtran.c rdswitch.c cdjpeg.c transupp.c
+jpegtran_SOURCES = jpegtran.c rdswitch.c cdjpeg.c
 jpegtran_LDADD   = libjpeg.la
 rdjpgcom_SOURCES = rdjpgcom.c
 wrjpgcom_SOURCES = wrjpgcom.c
--- jpeg-9e.orig/configure.ac
+++ jpeg-9e/configure.ac
@@ -7,6 +7,9 @@
 
 AC_INIT([libjpeg], [9.5.0])
 
+# Directory for additional local autotools macros.
+AC_CONFIG_MACRO_DIR([m4])
+
 # Directory where autotools helper scripts lives.
 AC_CONFIG_AUX_DIR([.])
 
